src/SFA.DAS.Authorization/AuthorizationContext.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System.Collections.Generic;
+
+namespace SFA.DAS.Authorization
+{
+    public class AuthorizationContext : IAuthorizationContext
+    {
+        private readonly Dictionary<string, object> _data = new Dictionary<string, object>();
+
+        public T Get<T>(string key)
+        {
+            if (_data.TryGetValue(key, out var value))
+            {
+                return (T)value;
+            }
+
+            throw new KeyNotFoundException($"The key '{key}' was not present in the dictionary");
```

Comment:
Minor point, but storing the context in a dictionary is an implementation detail, so the message could be more generic, such as '.. was not found in the context'.